### PR TITLE
[docs] Add xgboost_ray to docs

### PR DESCRIPTION
**Pull request by richardliaw**
## Why are these changes needed?

Make sure the package reference is being displayed properly.

## Checks

- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: doc/source/xgboost-ray.rst [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+
+``xgboost_ray`` provides a drop-in replacement for XGBoost's ``train``
+function. To pass data, instead of using ``xgb.DMatrix`` you will
+have to use ``xgboost_ray.RayDMatrix``.
+
+Here is a simplified example:
+
+.. code-block:: python
+
+    from xgboost_ray import RayDMatrix, train
```
Comment (amogkam) [inline doc/source/xgboost-ray.rst:34]: Should we create a python script and add it to the CI to make sure we pick up any API changes?

**Review by amogkam — COMMENTED** [review]
Left a comment about making the doc examples into python scripts and adding to the CI

**Comment by richardliaw** [comment]
hey @amogkam the additional tune file doesn't seem to be passing on Travis. Could you take a look?

**Comment by richardliaw** [comment]
awesome @amogkam looks like tests are passing, could you help approve?

File: python/ray/util/xgboost/BUILD [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+    deps = [":xgb_lib"],
+    tags = ["exlcusive"]
+)
+
+# This is a dummy test dependency that causes the above tests to be
+# re-run if any of these files changes.
+py_library(
+    name = "xgb_lib",
+    srcs = glob(["**/*.py"], exclude=["tests/*.py"]),
+)
```
Comment (amogkam) [inline python/ray/util/xgboost/BUILD:26]: Add new line here?

File: python/ray/util/xgboost/__init__.py [hunk 3]
```diff
@@ -0,0 +1,16 @@
+import logging
+
+logger = logging.getLogger(__name__)
+
+train = None
+predict = None
+RayDMatrix = None
```
Comment (amogkam) [inline python/ray/util/xgboost/__init__.py:7]: Also need to add `RayFileType=None`?

**Review by amogkam — APPROVED** [review]
This is great! Just left a couple small nits.

File: python/ray/util/xgboost/__init__.py (hunk 3 above)
Comment (richardliaw) [reply python/ray/util/xgboost/__init__.py:7]: oh good catch oops, didn't see this

**Pull request merged**